pkg/controllers/nodeclaim/garbagecollection/suite_test.go

```diff
@@ -332,3 +316,77 @@ var _ = Describe("GarbageCollection", func() {
 		})
 	})
 })
+
+var _ = Describe("NetworkInterface Garbage Collection", func() {
+	It("should not delete a network interface if a nodeclaim exists for it", func() {
+		// Create and apply a NodeClaim that references this NIC
+		nodeClaim := coretest.NodeClaim()
+		ExpectApplied(ctx, env.Client, nodeClaim)
+
+		// Create a managed NIC
+		nic := test.Interface(test.InterfaceOptions{Name: instance.GenerateResourceName(nodeClaim.Name), NodepoolName: nodePool.Name})
+		azureEnv.NetworkInterfacesAPI.NetworkInterfaces.Store(lo.FromPtr(nic.ID), *nic)
+
+		nicsBeforeGC, err := azureEnv.InstanceProvider.ListNics(ctx)
+		ExpectNoError(err)
+		Expect(len(nicsBeforeGC)).To(Equal(1))
+
+		// Run garbage collection
+		ExpectSingletonReconciled(ctx, networkInterfaceGCController)
+
+		// Verify NIC still exists after GC
+		nicsAfterGC, err := azureEnv.InstanceProvider.ListNics(ctx)
+		ExpectNoError(err)
+		Expect(len(nicsAfterGC)).To(Equal(1))
+	})
+	It("should delete a NIC if there is no associated VM", func() {
+		nic := test.Interface(test.InterfaceOptions{NodepoolName: nodePool.Name})
+		nic2 := test.Interface(test.InterfaceOptions{NodepoolName: nodePool.Name})
+		azureEnv.NetworkInterfacesAPI.NetworkInterfaces.Store(lo.FromPtr(nic.ID), *nic)
+		azureEnv.NetworkInterfacesAPI.NetworkInterfaces.Store(lo.FromPtr(nic2.ID), *nic2)
+		nicsBeforeGC, err := azureEnv.InstanceProvider.ListNics(ctx)
+		ExpectNoError(err)
+		Expect(len(nicsBeforeGC)).To(Equal(2))
+		// add a nic to azure env, and call reconcile. It should show up in the list before reconcile
+		// then it should not showup after
+		ExpectSingletonReconciled(ctx, networkInterfaceGCController)
+		nicsAfterGC, err := azureEnv.InstanceProvider.ListNics(ctx)
+		ExpectNoError(err)
+		Expect(len(nicsAfterGC)).To(Equal(0))
+	})
+	It("should not delete a NIC if there is an associated VM", func() {
+		managedNic := test.Interface(test.InterfaceOptions{NodepoolName: nodePool.Name})
+		azureEnv.NetworkInterfacesAPI.NetworkInterfaces.Store(lo.FromPtr(managedNic.ID), *managedNic)
+		managedVM := test.VirtualMachine(test.VirtualMachineOptions{Name: lo.FromPtr(managedNic.Name), NodepoolName: nodePool.Name})
+		azureEnv.VirtualMachinesAPI.VirtualMachinesBehavior.Instances.Store(lo.FromPtr(managedVM.ID), *managedVM)
+		ExpectSingletonReconciled(ctx, networkInterfaceGCController)
+		// We should still have a network interface here
+		nicsAfterGC, err := azureEnv.InstanceProvider.ListNics(ctx)
+		ExpectNoError(err)
+		Expect(len(nicsAfterGC)).To(Equal(1))
+
+	})
+	It("the vm gc controller should remove the nic if there is an associated vm", func() {
```

Comment:
Had to read the code to understand what it does; maybe belongs under VM GC test section? Or just a better description:

```suggestion
	It("the VM GC controller should also delete the associated NIC", func() {
```